packages/node_modules/@webex/internal-plugin-ediscovery/test/unit/spec/content.js

```diff
@@ -357,4 +549,145 @@ describe('EDiscovery Content API Tests', () => {
+      const mockCachedSpace = {spaceId: uuid};
+
+      const spaceIdMap = {
+        get: sinon.stub().withArgs(uuid).returns(mockCachedSpace),
+        set: sinon.stub(),
+        has: sinon.stub().returns(true)
+      };
+
+      webex.internal.ediscovery.contentSummaryBySpaceIdCache = {
+        get: sinon.stub().withArgs(uuid).returns(spaceIdMap),
+        set: sinon.stub(),
+        has: sinon.stub().returns(true),
+        clear: sinon.stub()
+      };
+      const result = webex.internal.ediscovery.getRemoteContentSummaryBySpaceId(url, uuid)
+        .then((res) => {
+          expect(res.body).to.equal(mockCachedSpace);
+          assert.callCount(webex.internal.ediscovery.contentSummaryBySpaceIdCache.has, 1);
+          assert.callCount(spaceIdMap.has, 1);
+          assert.callCount(spaceIdMap.get, 1);
+        });
+
+      return result;
+    });
+
+    it('GetRemoteContentSummaryBySpaceId populates contentSummaryBySpaceIdCache with space when retrieved', async () => {
+      const mockResponse = {body: {spaceId: uuid}};
+
+      webex.internal.ediscovery.request = sinon.stub().resolves(mockResponse);
+
+      const spaceIdMap = {
+        set: sinon.stub(),
+        has: sinon.stub().returns(false)
+      };
+
+      webex.internal.ediscovery.contentSummaryBySpaceIdCache = {
+        get: sinon.stub().withArgs(uuid).returns(spaceIdMap),
+        set: sinon.stub(),
+        has: sinon.stub().returns(true),
+        clear: sinon.stub()
+      };
+      const result = webex.internal.ediscovery.getRemoteContentSummaryBySpaceId(url, uuid)
+        .then((res) => {
+          expect(res).to.equal(mockResponse);
+          assert.callCount(webex.internal.ediscovery.contentSummaryBySpaceIdCache.has, 2);
+          assert.callCount(spaceIdMap.has, 1);
+          assert.calledWith(spaceIdMap.set, mockResponse.body.spaceId, mockResponse.body);
+        });
+
+      return result;
+    });
+
+    it('GetRemoteContentSummaryBySpaceId cannot write space to contentSummaryBySpaceIdCache without spaceId', async () => {
+      const mockResponse = {body: {}};
+
+      webex.internal.ediscovery.request = sinon.stub().resolves(mockResponse);
+
+      const spaceIdMap = {
+        set: sinon.stub(),
+        has: sinon.stub().returns(false)
+      };
+
+      webex.internal.ediscovery.contentSummaryBySpaceIdCache = {
+        get: sinon.stub().withArgs(uuid).returns(spaceIdMap),
+        set: sinon.stub(),
+        has: sinon.stub().returns(true),
+        clear: sinon.stub()
+      };
+      const result = webex.internal.ediscovery.getRemoteContentSummaryBySpaceId(url, uuid)
+        .then((res) => {
+          expect(res).to.equal(mockResponse);
+          assert.callCount(webex.internal.ediscovery.contentSummaryBySpaceIdCache.has, 2);
+          assert.callCount(spaceIdMap.has, 1);
+          assert.notCalled(spaceIdMap.set);
+        });
+
+      return result;
+    });
+  });
```

Comment:
Most of these tests are modified versions of the existing API tests, with the exception of the 'fails with a url that does not contain a report id' case